text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,708 @@
+{
+    sdp: String,
+    tracks: {
+        receivers: [
+            {
+                kind: "audio" | "video",
+                id: String,
+            }
+        ],
+        senders: [
+            {
+                kind: "audio" | "video",
+                id: String,
+                uuid: String,
+                metadata: Option<String>,
+                state: Option<{
+                    active: bool,
+                    screen: bool,
+                }>,
+            }
+        ],
+    }
+}
+```
+
+**_Response data_**:
+
+```
+{
+    sdp: String
+}
+```
+
+### 4.6.2 Room actions, event
+
+We can subscribe to peers event (joined, left, track added, track removed) and also can unsubscribe from it.
+
+#### 4.6.2.1 Subscribe to other peers event
+
+(Note that this action only works with `event.subscribe` manual mode.)
+
+**_Cmd:_** `room.peers.subscribe`
+
+**_Request Data:_**
+
+```
+{
+    peer_ids: [String],
+}
+```
+
+**_Response Data:_** None
+
+#### 4.6.2.2 Unsubscribe to other peers event
+
+(Note that this action only works with `subscribe` manual mode.)
+
+**_Cmd:_**: `room.peers.unsubscribe`
+
+**_Request Data:_**
+
+```
+{
+    peer_ids: [String],
+}
+
+```
+
+#### 4.6.2.3 Peer joined event
+
+**_Cmd:_**: `room.peers.{peer}.joined`
+
+**_Event data:_**:
+
+```
+{
+    metadata: Option<String>,
+}
+```
+
+**_Response Data:_**: None
+
+#### 4.6.2.3 Peer left event
+
+**_Cmd:_**: `room.peers.{peer}.left`
+
+**_Event data:_**: None
+
+**_Response Data:_**: None
+
+#### 4.6.2.3 Track added event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.added`
+
+**_Event data:_**:
+
+```
+{
+    metadata: Option<String>,
+    state: {
+        active: bool,
+        screen: bool,
+        simulcast: bool,
+    },
+}
+```
+
+#### 4.6.2.3 Track updated event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.updated`
+
+**_Event data:_**:
+
+```
+{
+    state: {
+        active: bool,
+    },
+}
+```
+
+#### 4.6.2.3 Track removed event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.removed`
+
+**_Event data:_**: None
+
+**_Response Data:_**: None
+
+### 4.6.3 Session actions, event
+
+#### 4.6.3.1 Disconnect
+
+**_Cmd:_**: `session.disconnect`
+
+**_Request data:_**: None
+
+**_Response:_**: None
+
+#### 4.6.3.2 Goaway event
+
+Goaway event is sent by the server in some cases:
+
+- The server is going to shutdown or restart.
+- The client lifetime is expired, this is useful in some video conference application where each client only has limited session time; example 1 hour.
+- The client is kicked by the server.
+
+In case of a server shutdown or restart, the client should reconnect by sending restart-ice request.
+
+**_Cmd:_**: `session.on_goaway`
+
+**_Event data:_**:
+
+```
+{
+    reason: "shutdown" | "kick",
+    message: Option<String>,
+    remain_seconds: Number,
+}
+```
+
+**_Response Data:_**: None
+
+`remain_seconds` is the time that the client has to reconnect to the server. If it's 0, the client should reconnect immediately.
```

Comment:
`reconnect_after_secs` or `reconnect_after`

Reply:
I'm not agree with this, remain_seconds with meaning "you only have xxx seconds to be served". But for more clearly, it can be changed to: 

`remain_seconds`: This represents the remaining time that the client will be served by the server. In the event that a client needs to reconnect, it should do so before the remain_seconds expire to avoid interrupting the client session.

Reply:
oh sorry my bad for misreading the purpose, it's ok then